addons/l10n_ar/models/account_journal.py

```diff
@@ -107,12 +107,11 @@ def _get_journal_codes(self):
                     'l10n_latam_use_documents')
     def _check_afip_configurations(self):
         """ Do not let to update journal if already have confirmed invoices """
-        arg_sale_journals = self.filtered(lambda x: x.company_id.country_id == self.env.ref('base.ar') and
-                                          x.type == 'sale' and x._origin.type == 'sale')
-        invoices = self.env['account.move'].search([('journal_id', 'in', arg_sale_journals.ids), ('state', '!=', 'draft')])
+        journals = self.filtered(lambda x: x.company_id.country_id == self.env.ref('base.ar') and x.type in ['sale', 'purchase'])
```

Comment:
why is it important for sale as well?  Not sure about the ._origin as well.

Reply:
The idea is to avoid changing the journal type to sale or purchase if it has validated invoices (or posted before). Before this change, for instance, you could change the journal from sale to purchase or backwards.
About the _origin only works in onchange methods so it has no effect here.

Reply:
I think the only case for purchase is the latam_use_documents, but that makes sense.  And indeed _origin makes no sense here.